terraform/functions.tf

```diff
@@ -515,3 +515,112 @@ module "lambda_function-cpfValidation" {
     }
   }
 }
+
+
+module "lambda_function-cpfCreateArchive" {
+  source  = "terraform-aws-modules/lambda/aws"
+  version = "4.2.0"
+
+  // Metadata
+  function_name = "${var.namespace}-cpfCreateArchive"
+  description   = "Reacts to SQS events and generates archive files of CSV's."
+
+  // Networking
+  vpc_subnet_ids         = null
+  vpc_security_group_ids = null
+  attach_network_policy  = false
+
+  // Permissions
+  role_permissions_boundary         = local.permissions_boundary_arn
+  attach_cloudwatch_logs_policy     = true
+  cloudwatch_logs_retention_in_days = var.log_retention_in_days
+  attach_policy_jsons               = length(local.lambda_default_execution_policies) > 0
+  number_of_policy_jsons            = length(local.lambda_default_execution_policies)
+  policy_jsons                      = local.lambda_default_execution_policies
+  attach_policy_statements          = true
+  policy_statements = {
+    AllowDownloadExcelObjects = {
+      effect = "Allow"
+      actions = [
+        "s3:GetObject",
+        "s3:HeadObject",
+      ]
+      resources = [
+        # Path: treasuryreports/{organization_id}/{reporting_period_id}/{filename}.csv
```

Comment:
In https://github.com/usdigitalresponse/cpf-reporter/issues/315, the path also includes a `{user.id}` field. @as1729 which path should be used?